src/library_ccall.js

```diff
@@ -133,7 +133,7 @@ mergeInto(LibraryManager.library, {
     argTypes = argTypes || [];
     // When the function takes numbers and returns a number, we can just return
     // the original function
-    var numericArgs = argTypes.every((type) => type === 'number');
+    var numericArgs = argTypes.every((type) => type === 'number' || type === 'boolean');
```

Comment:
Should be possible to do this for `bigint` as well

Reply:
I think the bigint change would be good, but it might be a litte more involved.  Lets get this fix landed first.

Can you update one of the existing tests to use this type in that argument list?

Reply:
Sure, I'll see to it once I got time

Reply:
Hi, I just pushed a new commit [`
df31083`](https://github.com/emscripten-core/emscripten/pull/17511/commits/df31083f7d94351b4063e97c28b829f455eac941) to do the test for the boolean argument type.

I saw a `print_bool` test function there, so I did not try to add another function to test for the *update one of the existing tests to use this type in that argument list* thing you've mentioned. If I did not get it right, please let me know _as I'm not quite confident of my understanding regarding your suggestion_.

And during my local testing, I've found that the macro `#if !ASSERTIONS` was stopping the code from reaching those lines inside of its closure. Well, I think those lines would do nothing wrong in a `ASSERTIONS` context, so I decided to remove it. Please let me know if it can (or potentially can) break something.

Reply:
The idea is that when this whole block is only active in release builds.. when ASSERTIONS are enabled we want to skip this fast path and always use the slow path below.   I think you should restore the `#if`/`#endif` and we can try to figure out what else need to change.  Most likely its the ccall function also needs to be aware of this new type of argument

Reply:
Sure, I've restored the `#if`/`#endif` in commit [`60b8212`](https://github.com/emscripten-core/emscripten/pull/17511/commits/60b821229786c63444994f4cf3c0161981e11457)